automation code of OCP-30285

zhouying7780 opened the pull request:
@jianzhangbjz Could you please help review ? thanks.

akostadinov commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,61 @@
+                        defer closeFile(emptyFile)
+			out, err = oc.Run("version").Args("--kubeconfig", "/tmp/emptykubeconfig").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("Client Version"))
```
Can we check version string with a regular expression or whatever kind of pattern matching? We have seen in the past many changes like adding.removing `v` in front of the version sting and other minor changes that broke version parsing.

btw for YAML format we also better check path and version sting format with pattern. Will save us a lot of potential troubles down the road.

Thank you.

zhouying7780 replied:
@akostadinov thanks for review , will try to complete all the request.

zhouying7780 replied:
@akostadinov  Have updated , could you please review again ? thanks .

zhouying7780 commented:
[root@dhcp-140-138 openshift-tests]# ./extended-platform-tests run-test "[cli]oc version[Conformance] oc version before login check version without login [Suite:openshift/conformance/parallel/minimal]"
I0529 17:27:41.540469     947 test_context.go:419] Tolerating taints "node-role.kubernetes.io/master" when considering if nodes are ready
May 29 17:27:41.574: INFO: Waiting up to 30m0s for all (but 100) nodes to be schedulable
May 29 17:27:42.741: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
May 29 17:27:43.469: INFO: 0 / 0 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
May 29 17:27:43.469: INFO: expected 0 pod replicas in namespace 'kube-system', 0 are Running and Ready.
May 29 17:27:43.469: INFO: Waiting up to 5m0s for all daemonsets in namespace 'kube-system' to start
May 29 17:27:43.714: INFO: e2e test version: v0.0.0-master+$Format:%h$
May 29 17:27:43.947: INFO: kube-apiserver version: v1.18.3+1bec683
May 29 17:27:44.185: INFO: Cluster IP family: ipv4
[BeforeEach] [Top Level]
  /home/gopath/src/github.com/openshift/openshift-tests/test/extended/util/test.go:60
[BeforeEach] [cli]oc version[Conformance]
  /home/gopath/src/github.com/openshift/openshift-tests/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:154
STEP: Creating a kubernetes client
[BeforeEach] [cli]oc version[Conformance]
  /home/gopath/src/github.com/openshift/openshift-tests/test/extended/util/client.go:111
May 29 17:27:47.077: INFO: configPath is now "/tmp/configfile453419244"
May 29 17:27:47.077: INFO: The user is now "e2e-test-oc-ntf5b-user"
May 29 17:27:47.078: INFO: Creating project "e2e-test-oc-ntf5b"
May 29 17:27:47.695: INFO: Waiting on permissions in project "e2e-test-oc-ntf5b" ...
May 29 17:27:47.934: INFO: Waiting for ServiceAccount "default" to be provisioned...
May 29 17:27:48.274: INFO: Waiting for ServiceAccount "deployer" to be provisioned...
May 29 17:27:48.611: INFO: Waiting for ServiceAccount "builder" to be provisioned...
May 29 17:27:48.951: INFO: Waiting for RoleBinding "system:image-pullers" to be provisioned...
May 29 17:27:49.424: INFO: Waiting for RoleBinding "system:image-builders" to be provisioned...
May 29 17:27:49.896: INFO: Waiting for RoleBinding "system:deployers" to be provisioned...
May 29 17:27:50.376: INFO: Project "e2e-test-oc-ntf5b" has been fully provisioned.
[It] check version without login [Suite:openshift/conformance/parallel/minimal]
  /home/gopath/src/github.com/openshift/openshift-tests/test/extended/cli/version.go:24
STEP: check version info with default KUBECONFIG
STEP: check version info without --kubeconfig,~/.kube/config,KUBECONFIG
STEP: check version info with empty kubeconfig
STEP: check version info with correct kubeconfig
closing
[AfterEach] [cli]oc version[Conformance]
  /home/gopath/src/github.com/openshift/openshift-tests/test/extended/util/client.go:102
May 29 17:27:54.102: INFO: Deleted {user.openshift.io/v1, Resource=users  e2e-test-oc-ntf5b-user}, err: <nil>
May 29 17:27:54.344: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthclients  e2e-client-e2e-test-oc-ntf5b}, err: <nil>
May 29 17:27:54.584: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthaccesstokens  ksYejIDyT22BebM88nnmGgAAAAAAAAAA}, err: <nil>
[AfterEach] [cli]oc version[Conformance]
  /home/gopath/src/github.com/openshift/openshift-tests/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:155
May 29 17:27:54.584: INFO: Waiting up to 7m0s for all (but 100) nodes to be ready
STEP: Destroying namespace "e2e-test-oc-ntf5b" for this suite.
May 29 17:27:55.815: INFO: Running AfterSuite actions on all nodes
May 29 17:27:55.815: INFO: Running AfterSuite actions on node 1

zhouying7780 commented:
@soltysh  Could you please help review ? thanks

soltysh commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,101 @@
+                        }         
+
+			g.By("check version info without --kubeconfig,~/.kube/config,KUBECONFIG") 
+                        kubeconfig :=exutil.KubeConfigPath()
```
You can still default to `$HOME/.kube/config` so you need to ensure one does not exist.

zhouying7780 replied:
@soltysh  I've updated , please help review again, thanks.

jianzhangbjz commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,101 @@
+		oc                     = exutil.NewCLI("oc", exutil.KubeConfigPath())
+	)
+
+	g.Describe("oc version before login", func() {
```
@zhouying7780 I'd suggest not use the `Describe` in a `Describe`.

jianzhangbjz commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,101 @@
+	)
+
+	g.Describe("oc version before login", func() {
+		g.It("check version without login", func() {
```
Please follow the [naming rules](https://docs.google.com/document/d/1k761p65J0Ig81hwZWaw73QJV6SBa7kPxbjuTEBlBbF4/edit#heading=h.nvqhnby7zbfg), thanks!

zhouying7780 commented:
@akostadinov @soltysh  I've resolved all the comments , please help review again, thanks.

jianzhangbjz commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+	exutil "github.com/openshift/openshift-tests/test/extended/util"
+)
+
+var _ = g.Describe("[cli]oc version[Conformance]", func() {
```
@zhouying7780 For the test case title, please follow the [naming rules](https://docs.google.com/document/d/1k761p65J0Ig81hwZWaw73QJV6SBa7kPxbjuTEBlBbF4/edit#heading=h.irspvef076fy), thanks!

jianzhangbjz commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+		oc                     = exutil.NewCLI("oc", exutil.KubeConfigPath())
+	)
+
+	g.It("oc version before login", func() {
```
As the same as the above.

zhouying7780 replied:
@jianzhangbjz Thanks , have updated.

jianzhangbjz commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+                defaultHomeDir,_ := os.UserHomeDir()
+                defaultConfig := defaultHomeDir + "/.kube/config"
+		if _, err := os.Stat(defaultConfig); err == nil {
+                        defer os.Rename(defaultHomeDir+"/.kube/configback",defaultHomeDir+"/.kube/config")
```
Replace `defaultHomeDir+"/.kube/config"` with `defaultConfig`?

zhouying7780 replied:
Yes , restore it.

jianzhangbjz replied:
yes, I mean if we can use `os.Rename(defaultHomeDir+"/.kube/configback", defaultConfig)`.

zhouying7780 replied:
@jianzhangbjz The value of  'defaultConfig' have updated to 'defaultHomeDir+"/.kube/configback"'. so can't use it again.

kuiwang02 commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+	exutil "github.com/openshift/openshift-tests/test/extended/util"
+)
+
+var _ = g.Describe("[sig-apps]oc should[Conformance]", func() {
```
For the `Describe` field of your ginkgo test case, please add the "[sig-team] sub-team" label, for example g.Describe("[sig-operators] **OLM** should", func(), here OLM is sub-team

jianzhangbjz replied:
I guess the `oc` is the sub-team name. @zhouying7780 Correct me if I'm wrong, thanks!

soltysh replied:
sig-cli is the preferred one

zhouying7780 replied:
@soltysh @jianzhangbjz @kuiwang02   Thanks you all , have updated , please help review .

zhouying7780 replied:
@kuiwang02  thanks , the sub-team is Workloads. I've updated. Please help review again.

kuiwang02 commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+	exutil "github.com/openshift/openshift-tests/test/extended/util"
+)
+
+var _ = g.Describe("[sig-cli] oc [Slow]", func() {
```
HI, Per https://docs.google.com/document/d/1k761p65J0Ig81hwZWaw73QJV6SBa7kPxbjuTEBlBbF4/edit# name rules section, there is sub-team here. is oc the sub-team name? if it is, OK. if it is not, please add sub-team. thanks.

and should no test label [Slow] here, so please remove it. And, move it to g.It

kuiwang02 commented on test/extended/cli/version.go:
```diff
@@ -0,0 +1,106 @@
+		oc                     = exutil.NewCLI("oc", exutil.KubeConfigPath())
+	)
+
+	g.It("Critical-30285-Checking oc version before login", func() {
```
Please add [Slow] here. "Critical-30285-Checking oc version before login" --> "Critical-30285-Checking oc version before login [Slow]"

zhouying7780 replied:
@kuiwang02 Thanks , have updated , please help review again.

kuiwang02 replied:
@zhouying7780 it is good for me to move test label Slow. thanks

zhouying7780 replied:
@kuiwang02 Since the case will update KUBECONFIG and ~/.kube/config, so need to run by sequentially, I removed the [Slow] label and add [Serial]

zhouying7780 commented:
Latest test result:  http://pastebin.test.redhat.com/881528

zhouying7780 commented:
latest test result: http://pastebin.test.redhat.com/881539

kuiwang02 commented:
/lgtm

jianzhangbjz commented:
/approve

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/cherry-pick release-4.5
